### PR TITLE
Feature/projects crud #9

### PR DESCRIPTION
**Pull request by rekloyd**
#9 Added:

- Project Repository to be able to do CRUD operations.
- Project Input and Update project input to manage the body params in the requests.
- Project Service with all the business logic.
- Project Controller with all the project endpoints: create a project, get projects, update project, delete a project.

### CODE REVIEW

**Review by Jaranag — APPROVED** [review]
Great coding guys! 🔢

**Review by JoseAlfredoMiguel — APPROVED** [review]

File: src/main/java/com/grupo2/kanbanboard/controllers/ProjectController.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import com.grupo2.kanbanboard.entities.Project;
+import com.grupo2.kanbanboard.requests.CreateProjectInput;
+import com.grupo2.kanbanboard.requests.UpdateProjectInput;
+import com.grupo2.kanbanboard.services.ProjectService;
+
+@RestController
+public class ProjectController {
+    public ProjectService projectService;
+
+    public ProjectController(ProjectService projectService) {
```
Comment (SaraVictorFernandez) [inline src/main/java/com/grupo2/kanbanboard/controllers/ProjectController.java:25]: Very nice! Do you know how this pattern is called? (I'm referring to passing the ProjectService in the constructor)

File: src/main/java/com/grupo2/kanbanboard/requests/CreateProjectInput.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.grupo2.kanbanboard.requests;
+
+import com.grupo2.kanbanboard.entities.Project;
+
+public record CreateProjectInput(String name) {
+    public Project toProject() {
```
Comment (SaraVictorFernandez) [inline src/main/java/com/grupo2/kanbanboard/requests/CreateProjectInput.java:6]: I'm not sure about a request object knowing about the model

File: src/main/java/com/grupo2/kanbanboard/services/ProjectService.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+@Service
+public class ProjectService {
+    private final ProjectRepository projectRepository;
+
+    public ProjectService(ProjectRepository projectRepository) {
+        this.projectRepository = projectRepository;
+    }
+
+    public Project create(Project project) {
```
Comment (SaraVictorFernandez) [inline src/main/java/com/grupo2/kanbanboard/services/ProjectService.java:20]: if the project already exists... what this method is doing is not creating a project, but saving it

File: src/main/java/com/grupo2/kanbanboard/services/ProjectService.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+        return projects;
+    }
+
+    public Optional<Project> findById(int id) {
+        return projectRepository.findById(id);
+    }
+
+    public Project update(Project projectToUpdate) {
+        return projectRepository.save(projectToUpdate);
```
Comment (SaraVictorFernandez) [inline src/main/java/com/grupo2/kanbanboard/services/ProjectService.java:36]: By doing this, you are forcing the client to send all of the field for the project every time. More than an update this feels like an override

**Review by JiajunQiuWu — APPROVED** [review]

File: src/main/java/com/grupo2/kanbanboard/controllers/ProjectController.java (hunk 1 above)
Comment (rekloyd) [reply src/main/java/com/grupo2/kanbanboard/controllers/ProjectController.java:25]: Singleton pattern? (Joel's idea, if he's wrong we should fire him)

File: src/main/java/com/grupo2/kanbanboard/requests/CreateProjectInput.java (hunk 2 above)
Comment (rekloyd) [reply src/main/java/com/grupo2/kanbanboard/requests/CreateProjectInput.java:6]: Why?

**Review by JiajunQiuWu — APPROVED** [review]

**Review by JoseAlfredoMiguel — APPROVED** [review]

**Review by SaraVictorFernandez — APPROVED** [review]
lgtm!

**Pull request merged**